swagger.yaml

```diff
@@ -10,33 +10,322 @@ basePath: "/v1"
 schemes:
   - http
 tags:
-  - name: "hello"
+  - name: "roles"
+  - name: "permissions"
+  - name: "policies"
 paths:
-  /hello:
+
+  /roles:
+    get:
+      tags:
+        - "roles"
+      summary: "Returns a list of all roles"
+      description: "Returns a paginated list of all roles" 
+      produces:
+        - "application/json"
+      responses:
+        200: 
+          description: "Successfully returned a json object containing a list of all roles"
+          schema:
+            type: object
+            required: 
+              - items
+            properties: 
+              count:
+                type: integer
+                readOnly: true
+                description: "The number of roles returned"
+              total_count:
+                type: integer
+                readOnly: true
+                description: "The total number of roles"
+              offset:
+                type: integer
+                readOnly: true
+                description: "The first row of resources to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+              limit:
+                type: integer
+                readOnly: true
+                description: "The number of roles per page"
+              items:
+                description: "A list of roles"
+                type: array
+                items:
+                  $ref: "#/definitions/Role"
+        400:
+          description: "Invalid request"
+        500:
+          $ref: "#responses/InternalError"
```

Comment:
I think this one should be `$ref: "#/responses/InternalError"` (with / after #). There are a couple of typos. What I do is copy the whole file and paste into https://editor.swagger.io/ - the right hand side will show you where the errors are

Reply:
Thank you for spotting this, I will update the PR.

Reply:
Just pushed changes to address these typos